8-PLIER_pathways_analysis.Rmd

```diff
@@ -0,0 +1,115 @@
+```
+
+### What additional pathways are identified by PLIER after doubling the sample size?
+
+Here we look for oncogenic pathways (defined by [GSEA MSigDB C6](https://www.gsea-msigdb.org/gsea/msigdb/human/genesets.jsp?collection=C6)) that are found more frequently in the data with the full sample size than in data with half the sample size.
+By focusing on pathways that are reliably detected more often in the full sample size data, we can identify stable patterns that emerge when more data is available.
+We ran PLIER `r n_repeats` times using each combination of data normalization method and sample size (half or full).
+Out of those `r n_repeats` runs, we found the proportion of times each oncogenic pathway was significantly associated with a latent variable.
+We set an arbitrary "meaningful difference" threshold of `r meaningful_difference` to detect when that proportion was meaningfully greater using full sample size compared to half sample size and require array and RNA-seq data to both to satisfy the condition.
+We also arbitrarily limit results to those pathways found in over half of runs using the full size data.
+
+### Load packages and pathways
+
+```{r}
+library(tidyverse)
+data("oncogenicPathways", package = "PLIER")
+```
+
+
+### Set input file names
+```{r}
+plots_data_dir <- here::here("plots", "data")
+brca_filename <- file.path(plots_data_dir, "BRCA_subtype_PLIER_pathways.tsv")
+gbm_filename <- file.path(plots_data_dir, "GBM_subtype_PLIER_pathways.tsv")
+```
+
+### Read in data for each cancer type
+```{r}
+pathways_df <- NULL
+
+if (file.exists(brca_filename)) {
+  brca_pathways_df <- read_tsv(brca_filename) %>%
+    mutate(cancer_type = "BRCA")
+  pathways_df <- bind_rows(pathways_df,
+                           brca_pathways_df)
+} else {
+  message(str_c("BRCA file ", brca_filename, " does not exist."))
+}
+
+if (file.exists(gbm_filename)) {
+  gbm_pathways_df <- read_tsv(gbm_filename) %>%
+    mutate(cancer_type = "GBM")
+  pathways_df <- bind_rows(pathways_df,
+                           gbm_pathways_df)
+} else {
+  message(str_c("GBM file ", gbm_filename, " does not exist."))
+}
+
+if (!file.exists(brca_filename) & !file.exists(gbm_filename)) {
+  stop(str_c("Neither BRCA file ", brca_filename,
+             " nor GBM file ", gbm_filename,
+             " exists."))
+}
+
+```
+
+### Filter data to identify oncogenic pathways detected more frequently in full data
+```{r}
+
+pathways_df %>%
+  filter(FDR < 0.05, # require significant association with an LV
+         pathway %in% colnames(oncogenicPathways)) %>% # require oncogenic pathways
+  # for each combination of normalization method, %RNA-seq, pathway, and cancer type
+  group_by(nmeth, pseq, pathway, cancer_type) %>%
+  # summarize by finding the count and proportion of repeats in which that
+  # pathway was significantly associated with at least one latent variable
+  summarize(n_seeds = length(unique(seed_index)),
+            prop_seeds = n_seeds/n_repeats,
```

Comment:
Checking my understanding here.

To jog my memory about how this is set up, each repeat gets its own seed, so the total number of seeds to start will be equal to the number of repeats? 

Then only the seeds that meet the FDR < 0.05 threshold of this method, % sequencing data, and pathway combination will be counted in `n_seeds`.

Reply:
Yes precisely: `n_seeds` is the number of repeats where the combination of pathway/method/%seq is significant for one or more latent variables